### PR TITLE
Felix/realsense sim bringup

### PR DESCRIPTION
**Pull request by FelixKHWong**
Integrating the Realsense camera into Sim so that the Camera transforms are published and so we can see things in gazebo.
Currently all camera streams are online but the Sim performance does suffer with around 8 fps. The point cloud looks good and the camera stream makes sense. Able to move robot and see camera update which is cool. Also can see arms moving in front and the planned path from the camera's view.

See this link for some more context. https://bdschnap.atlassian.net/wiki/spaces/FYDP/pages/34635777/Simulation
This plugin also requires the PAL Realsense plugin to be downloaded and built once with catkin. https://github.com/pal-robotics/realsense_gazebo_plugin

To turn off the point cloud and improve performance go to movo_common/movo_description/urdf/sensors/_d435.gazebo.xacro and set to be false

Below is a sample sim use. The camera stream in the bottom left has the lens in frame but those have since been removed by moving the optical joints 0.005m forward.
camera_sim_2

since two different urdf files are used, one for sim and one for real life, the environment variable MOVO_SIM must be set to true (export MOVO_SIM=true) to enable the camera in SIM. Otherwise, the model will default to the real life camera

### CODE REVIEW

**Comment by FelixKHWong** [comment]
Will add devnote update on how to download the necessary files and build them in next commit

**Comment by FelixKHWong** [comment]
Added the laser scan merger and the scans from front laser and camera laser can be seen. However, seems like there is some other extrapolation or filtering so I'm not sure if the closest point is always seen.
Here is the scan from the camera
camera_scan

Here is the scan from the front laser 
front_scan

Here is the combined scan on base_scan_filtered. The frequency is 4hz for the combined and camera while the front scan publishes at around 14 hz
base_scan_filtered

**Comment by FelixKHWong** [comment]
@younesr1 Could I get a review for this one? I think it's straight forward and doesn't impact other things in sim much other than it gives us simulated camera

File: dev_notes.txt [hunk 1]
```diff
@@ -5,8 +5,10 @@ movo2 can be reached by `ssh robohub@movo2`
 
 # Camera
 Camera uses the realsense-ros pkg
-Can be launched with <roslaunch movo_bringup movo2.launch as part of movo>
+Can be launched with <roslaunch movo_bringup movo2.launch> as part of movo
 Can be launched with <roslaunch realsense2_camera rs_camera.launch filters:=pointcloud>
+Camera Sim requires the clone of the PAL realsense plugin https://github.com/pal-robotics/realsense_gazebo_plugin into the workspace/src directory. Then build the workspace to compile the librealsense_gazebo_plugin.so.
```
Comment (nightingale-project) [inline dev_notes.txt]: can u add this to the rosinstall and remove this comment?

File: movo_common/movo_description/urdf/movo_components/movo_pan_tilt.urdf.xacro [hunk 2]
```diff
@@ -26,8 +26,17 @@ LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
 NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS 
 SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 -->
+
 <robot name="movo_pan_tilt" xmlns:xacro="http://ros.org/wiki/xacro">
-    <xacro:include filename="$(find movo_description)/urdf/sensors/_d435.urdf.xacro"/>
+    <!-- swap out these d435 realsense files for sim (_d435_sim) and for real.  use "export MOVO_SIM=true" to set sim as environment variable-->
```
Comment (nightingale-project) [inline movo_common/movo_description/urdf/movo_components/movo_pan_tilt.urdf.xacro]: would it be better if this took in the arg indicating if this is sim or not (the arg already exists, its essential for us to not launch nodes that interface with HW in sim)

Comment (FelixKHWong) [reply movo_common/movo_description/urdf/movo_components/movo_pan_tilt.urdf.xacro]: Yeah a command line variable does sound better. Where is the argument located? I can try to pass it through to the file

Comment (FelixKHWong) [reply movo_common/movo_description/urdf/movo_components/movo_pan_tilt.urdf.xacro]: ok I think I got it. I initialize a "sim" arg that is only ever set to "true" in the simulation launch file and is false everywhere else. When running sim, the parameter is passed through down to the xacro and should always use the HW unless the movo_sim/movo.launch file is called when running sim

**Review by younesr1 — COMMENTED** [review]
checked the arg passing logic. looks right

**Review by younesr1 — APPROVED** [review]

**Comment by FelixKHWong** [comment]
@younesr1 do you need to go onto the nightingale account to also approve this?

**Review by younesr1 — APPROVED** [review]

**Pull request merged**